astrocut/make_cube.py

```diff
@@ -346,3 +349,496 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
             print(f"Total time elapsed: {(time() - startTime)/60:.2f} min")
 
         return self.cube_file
+
+
+class TicaCubeFactory():
+
+    def __init__(self, max_memory=50):
+        """ Setting up the class members. NOTE: Later down the line 
+        we might want to generalize CubeFactory so that we no longer need 
+        a separate class for the TICA cubes. 
+        
+        Modifying the __init__ 
+        in CubeFactory so that it can switch parameters based on whether 
+        we want cubes for TICA or TESS will probably be the route I take.
+        Or see how this is generalized when Zcut wraps CubeFactory? 
+        """
+
+        self.max_memory = max_memory  # in GB
+        self.block_size = None  # Number of rows
+        self.num_blocks = None
+        self.cube_shape = None
+        
+        self.time_keyword = 'STARTTJD'  # Start time in TJD. TICA-specific.
+        self.last_file_keywords = ['ENDTJD']  # Stop time in TJD. TICA-specific (assumed to be in extension 0)                  
+        self.image_header_keywords = ['CAMNUM', 'CCDNUM']  # Camera number and CCD number 
+        self.template_requirements = {'NAXIS': 2}  # Using NAXIS instead of WCSAXES. JENNY: Verify this is OK.
+                                                   
+        self.file_list = None
+        self.template_file = None
+        
+        self.primary_header = None
+        self.info_table = None
+        self.cube_file = None
+
+        # Used when updating an existing cube 
+        self.old_cols = None
+        self.update = False
+        self.cube_append = None
+
+    def _configure_cube(self, file_list, **extra_keywords):
+        """ Run through all the files and set up the  basic parameters for the cube.
+        Set up the cube primary header.
+        """
+
+        file_list = np.array(file_list)
+        image_shape = None
+        start_times = np.zeros(len(file_list))
+        for i, ffi in enumerate(file_list):
+            ffi_data = fits.open(ffi, mode='denywrite', memmap=True)
+            
+            start_times[i] = ffi_data[0].header.get('STARTTJD')
+
+            if image_shape is None:  # Only need to fill this once
+                image_shape = ffi_data[0].data.shape
```

Comment:
Since the ffi_data order is diff. between TICA and normal, we might want to set these indices as class attributes.

(image data is 0 for TICA but 1 for the original ones)

Reply:
I'm going to do another PR to add class attributes to the original CubeFactory so we can have 1 analogous CubeFactory for SPOC and TICA, and I'll add an indexing class attribute as well.

Reply:
is there an issue for this?

Reply:
Yes: https://github.com/spacetelescope/astrocut/issues/62